tests/components/zoneminder/test_init.py

```diff
@@ -0,0 +1,122 @@
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.async_mock import MagicMock, patch
+from tests.common import async_fire_time_changed
+
+
+async def test_no_yaml_config(hass: HomeAssistant) -> None:
+    """Test empty yaml config."""
+    with patch(
+        "homeassistant.components.zoneminder.common.ZoneMinder", autospec=ZoneMinder
+    ) as zoneminder_mock:
+        zm_client: ZoneMinder = MagicMock(spec=ZoneMinder)
+        zm_client.get_zms_url.return_value = "http://host1/path_zms1"
+        zm_client.login.return_value = True
+        zm_client.get_monitors.return_value = []
+
+        zoneminder_mock.return_value = zm_client
+
+        hass_config = {const.DOMAIN: []}
+        await async_setup_component(hass, const.DOMAIN, hass_config)
+        await hass.async_block_till_done()
+        assert not hass.services.has_service(const.DOMAIN, const.SERVICE_SET_RUN_STATE)
+
+
+async def test_yaml_config_import(hass: HomeAssistant) -> None:
+    """Test yaml config import."""
+    with patch(
+        "homeassistant.components.zoneminder.common.ZoneMinder", autospec=ZoneMinder
+    ) as zoneminder_mock:
+        zm_client: ZoneMinder = MagicMock(spec=ZoneMinder)
+        zm_client.get_zms_url.return_value = "http://host1/path_zms1"
+        zm_client.login.return_value = True
+        zm_client.get_monitors.return_value = []
+
+        zoneminder_mock.return_value = zm_client
+
+        hass_config = {const.DOMAIN: [{CONF_HOST: "host1"}]}
+        await async_setup_component(hass, const.DOMAIN, hass_config)
+        await hass.async_block_till_done()
+        assert hass.services.has_service(const.DOMAIN, const.SERVICE_SET_RUN_STATE)
+
+
+async def test_load_call_service_and_unload(hass: HomeAssistant) -> None:
+    """Test config entry load/unload and calling of service."""
+    with patch(
+        "homeassistant.components.zoneminder.common.ZoneMinder", autospec=ZoneMinder
+    ) as zoneminder_mock:
+        zm_client: ZoneMinder = MagicMock(spec=ZoneMinder)
+        zm_client.get_zms_url.return_value = "http://host1/path_zms1"
+        zm_client.login.side_effect = [True, True, False, True]
+        zm_client.get_monitors.return_value = []
+        zm_client.is_available.return_value = True
+
+        zoneminder_mock.return_value = zm_client
+
+        await hass.config_entries.flow.async_init(
+            const.DOMAIN,
+            context={CONF_SOURCE: config_entries.SOURCE_USER},
+            data={
+                CONF_HOST: "host1",
+                CONF_USERNAME: "username1",
+                CONF_PASSWORD: "password1",
+                CONF_PATH: "path1",
+                const.CONF_PATH_ZMS: "path_zms1",
+                CONF_SSL: False,
+                CONF_VERIFY_SSL: True,
+            },
+        )
+        await hass.async_block_till_done()
+
+        config_entry = next(iter(hass.config_entries.async_entries(const.DOMAIN)), None)
+        assert config_entry
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)